src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -1545,7 +1545,16 @@ void uct_dc_mlx5_ep_handle_failure(uct_dc_mlx5_ep_t *ep, void *arg,
     uct_dc_mlx5_update_tx_res(iface, txwq, txqp, pi);
     uct_rc_txqp_purge_outstanding(&iface->super.super, txqp, ep_status, pi, 0);
 
+    /* Invoke a user's error callback and release TX/FC resources before
+     * releasing DCI to have DCI for doing possible flush(CANCEL) */
     ucs_assert(ep->dci != UCT_DC_MLX5_EP_NO_DCI);
+    uct_dc_mlx5_iface_set_ep_failed(iface, ep, cqe, txwq, ep_status);
+
+    if (ep->dci != UCT_DC_MLX5_EP_NO_DCI) {
+        /* Try to return DCI into iface stack */
+        uct_dc_mlx5_iface_dci_put(iface, dci_index);
+    }
+
```

Comment:
looks weird, duplication of lines 1559-1560

Reply:
removal of these lines wasn't added to commit by some reason
fixed